### PR TITLE
Inline TLS access.

### PR DESCRIPTION
**Pull request by jaykrell**
The same functions are referenced by JIT and runtime.
The JIT uses cannot be inlined. The runtime can.
Since I could not get anything else to build and the change is simple and mechanical, now two sets of functions are provided, inline and extern. Extern just calls inline.
This PR appears to change all the uses, but it is only the JIT uses. The runtime ones just work and keep the old names.

We should also consider building desktop with LTO/LTCG, which would do all this automatically, smarter, more generally. A quick experiment shows it grows the runtime size, which is why I say desktop.

### CODE REVIEW

**Comment by jaykrell** [comment]
Huh? This somehow trips the wasm-ld bug. Will retry rebase to master or something.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/utils/mono-tls.h [hunk 1]
```diff
@@ -101,18 +110,190 @@ mono_native_tls_set_value (MonoNativeTlsKey key, gpointer value)
...
-void mono_tls_set_lmf_addr 	   (MonoLMF           **value);
+
+G_EXTERN_C MonoInternalThread *mono_tls_get_thread_extern (void);
+G_EXTERN_C MonoJitTlsData     *mono_tls_get_jit_tls_extern (void);
+G_EXTERN_C MonoDomain *mono_tls_get_domain_extern (void);
+G_EXTERN_C SgenThreadInfo *mono_tls_get_sgen_thread_info_extern (void);
+G_EXTERN_C MonoLMF       **mono_tls_get_lmf_addr_extern (void);
+
+/*
+ * On all platforms we should be able to use either __thread or __declspec (thread)
```
Comment (lambdageek) [inline mono/utils/mono-tls.h]: Did you mean to also leave the original copy of these paragraphs in `mono-tls.c`

Comment (jaykrell) [reply mono/utils/mono-tls.h]: I meant to move almost everything to the .h.
I kinda didn't think about where is most readable.
And larger .h is bad for compile perf perhaps, at least it used to be.
I can look at it again and try to decide what is most readable.

I'd really like to remove the non-default `__thread` stuff, but I kept it.

**Pull request merged**